src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -921,45 +921,55 @@ public static bool IsValidParams(Symbol member)
-        private static bool IsOverride(Symbol overridden, Symbol overrider)
+        /// <summary>
+        /// Does <paramref name="moreDerivedOverride"/> override <paramref name="member"/> or the
+        /// thing that it originally overrides, but in a more derived class?
+        /// </summary>
+        private static bool IsMoreDerivedOverride(Symbol member, Symbol moreDerivedOverride)
         {
-            if (TypeSymbol.Equals(overridden.ContainingType, overrider.ContainingType, TypeCompareKind.ConsiderEverything2) ||
-                !MemberSignatureComparer.SloppyOverrideComparer.Equals(overridden, overrider))
+            if (!moreDerivedOverride.IsOverride ||
+                !IsBaseClass(derivedType: moreDerivedOverride.ContainingType, baseType: member.ContainingType) ||
+                !MemberSignatureComparer.SloppyOverrideComparer.Equals(member, moreDerivedOverride))
             {
                 // Easy out.
                 return false;
             }
 
-            // Does overrider override overridden?
-            var current = overrider;
-            while (true)
-            {
-                if (!current.IsOverride)
-                {
-                    return false;
-                }
-                current = current.GetOverriddenMember();
+            // Rather than following the member.GetOverriddenMember() chain, we check to see if both
+            // methods ultimately override the same original method.  This addresses issues in binary compat
+            // scenarios where the override chain may skip some steps.
+            // See https://github.com/dotnet/roslyn/issues/45798 for an example.
+            return moreDerivedOverride.GetLeastOverriddenMember(accessingTypeOpt: null).OriginalDefinition ==
+                   member.GetLeastOverriddenMember(accessingTypeOpt: null).OriginalDefinition;
+        }
 
-                // We could be in error recovery.
-                if ((object)current == null)
-                {
-                    return false;
-                }
+        private static bool IsBaseClass(TypeSymbol derivedType, TypeSymbol baseType)
+        {
+            Debug.Assert((object)derivedType != null);
+            Debug.Assert((object)baseType != null);
 
-                if (current == overridden)
-                {
-                    return true;
-                }
+            // A base class has got to be a class. The derived type might be a struct, enum, or delegate.
+            if (!baseType.IsClassType())
+            {
+                return false;
+            }
 
-                // Don't search beyond the overridden member.
-                if (TypeSymbol.Equals(current.ContainingType, overridden.ContainingType, TypeCompareKind.ConsiderEverything2))
+            for (TypeSymbol b = derivedType.BaseTypeNoUseSiteDiagnostics; (object)b != null; b = b.BaseTypeNoUseSiteDiagnostics)
+            {
+                if (b.Equals(baseType, TypeCompareKind.ConsiderEverything))
```

Comment:
>ConsiderEverything [](start = 55, length = 18)

Why ```ConsiderEverything```?

Reply:
Because methods in a method group are gathered from the inheritance hierarchy of the receiver, and therefore the receiver types must be related by precise inheritance.

---
In reply to: [461758371](https://github.com/dotnet/roslyn/pull/46367#discussion_r461758371) [](ancestors = 461758371)

Reply:
> Because methods in a method group are gathered from the inheritance hierarchy of the receiver, and therefore the receiver types must be related by precise inheritance.

This component is not responsible to gather candidates and doesn't control how it is done. I think we should not make any assumptions about what could be in the set. I think that Either we should come up with a test demonstrating that usage of ```ConsiderEverything``` is significant for correctness, otherwise we should ignore all options, which will be more robust approach in the log run. 

---
In reply to: [461827656](https://github.com/dotnet/roslyn/pull/46367#discussion_r461827656) [](ancestors = 461827656,461758371)

Reply:
This code isn't making an assumption.  It is checking it.  This parallels the existing behavior of overload resolution (see line 949 of the old code - an exact check).  If you don't understand how this is modeling the language specification please let me know and I will explain in detail.

---
In reply to: [461938115](https://github.com/dotnet/roslyn/pull/46367#discussion_r461938115) [](ancestors = 461938115,461827656,461758371)